rbac/internal/views.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for internal tenant management."""
+import datetime
+import json
+import logging
+
+import pytz
+from django.conf import settings
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from management.models import Group, Role
+from tenant_schemas.utils import tenant_context
+
+from api.models import Tenant
+
+
+logger = logging.getLogger(__name__)
+
+
+def destructive_ok():
+    """Determine if it's ok to run destructive operations."""
+    now = datetime.datetime.utcnow().replace(tzinfo=pytz.UTC)
+    return now < settings.INTERNAL_DESTRUCTIVE_API_OK_UNTIL
+
+
+def tenant_is_unmodified():
```

Comment:
This _could_ be more database efficient, but it's not really a big deal... Here you've got 4 db queries regardless of whether it's a modified or unmodified tenant.

Reply:
Just pushed a change to short-circuit the counts as they check.